frontend/src/vendor/list/VendorCardList.js

```diff
@@ -7,22 +7,17 @@ import SearchArea from "../../components/SearchArea";
 import ErrorBody from "../../components/ErrorBody";
 import sortByName from "../sort/sortByName";
 import filteredArrayByKeyword from "../filter";
+import { useQuery } from "react-query";
 
 const VendorCardList = () => {
   const [searchTerm, setSearchTerm] = useState("");
-  const [vendors, setVendors] = useState([]);
-  const [error, setError] = useState(null);
+  const {
+    data: vendors,
+    isLoading,
+    isError,
+  } = useQuery("vendors", fetchVendors);
 
-  const getVendors = async () => {
-    const result = await fetchVendors();
-    setVendors(result);
-  };
-
-  useLayoutEffect(() => {
-    getVendors().catch((err) => setError(err.message));
-  }, []);
-
-  vendors.sort(sortByName);
+  !isLoading && vendors.sort(sortByName);
```

Comment:
Ez nem túl szép megoldás, sőt kerülendő. Vagy kifejezést vagy utasítást írunk, a kettőt nem szerencsés keverni, mert nehezen látszik mi történik. A vendors.sort egy utasítás, aminek van mellékhatása, az egész sor viszont egy kifejezés a logikai ! és && miatt. Az, hogy a vendors.sort nem fut le, ha az isLoading true egy ügyes megfigyelés, de az olvasó számára nem látszik a szándék. Első ránézésre olyan, mintha egy logikai kifejezést értékelnénk ki, de a végeredményét nem is használjuk, emiatt egy óvatlan refactor során lehet ki is törölném.

A megoldás csupán annyi, hogy egy if-be kellene átalakítani, hogy a kód kifejezze a szándékot. De a szándék valójában amúgy nem is az, hogy a töltés függvényében akarod rendezni a vendors tömböt, hanem valójában egy null-checket akarsz, így a legegyszerűbb az optional chaining lenne: vendors?.sort(sortByName);